x-pack/platform/plugins/shared/dataset_quality/server/rule_types/context.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { getHumanReadableComparator } from '@kbn/stack-alerts-plugin/common';
+import type { Comparator } from '@kbn/stack-alerts-plugin/common/comparator_types';
+import { DatasetQualityRuleParams } from './types';
+
+const generateTitle = (ruleName: string, group: string, isRecovered = false) =>
+  i18n.translate('xpack.datasetQuality.rule.alertTypeContextTitle', {
+    defaultMessage: `rule ''{name}'' {verb}`,
+    values: {
+      name: ruleName,
+      verb: isRecovered ? 'recovered' : `matched query for group ${group}`,
+    },
+  });
+
+const generateMessage = (
+  value: string,
+  group: string,
+  timeSize: number,
+  timeUnit: string,
+  comparator: Comparator,
+  threshold: number[]
+) =>
+  i18n.translate('xpack.datasetQuality.rule.alertTypeContextReasonDescription', {
+    defaultMessage: `Degraded documents percentage is {value} in the last {window} for {group}. Alert when {comparator} {threshold}.`,
+    values: {
+      value,
+      window: `${timeSize}${timeUnit}`,
+      group,
+      comparator: getHumanReadableComparator(comparator),
+      threshold: threshold.join(' and '),
+    },
+  });
+
+const generateConditions = (
+  group: string,
+  comparator: Comparator,
+  threshold: number[],
+  isRecovered = false
+) =>
+  i18n.translate('xpack.datasetQuality.rule.alertTypeContextConditionsDescription', {
+    defaultMessage:
+      'Degraded documents percentage for {group} is {negation}{thresholdComparator} {threshold}',
+    values: {
+      group,
+      thresholdComparator: getHumanReadableComparator(comparator),
+      threshold: threshold.join(' and '),
+      negation: isRecovered ? 'NOT ' : '',
+    },
+  });
+
+export const generateContext = (
+  ruleName: string,
+  group: string,
+  date: string,
+  value: string,
+  params: DatasetQualityRuleParams,
+  isRecovered = false
+) => ({
+  title: generateTitle(ruleName, group, isRecovered),
+  date,
+  group,
```

Comment:
You should probably check with @benakansara regarding context values for "grouping" info; there is work underway to standardize how these work, mainly for multi-level groupings, but probably applicable to any rule that can "group".  For example, see: https://github.com/elastic/kibana/pull/213550

Reply:
Addressed in [0cd3510](https://github.com/elastic/kibana/pull/216026/commits/0cd3510c9049cdd3824e31aca72cd699c4bff014).